packages/nx/src/utils/package-json.ts

```diff
@@ -134,16 +134,21 @@ export function buildTargetFromScript(
   };
 }
 
-export function readTargetsFromPackageJson({ scripts, nx }: PackageJson) {
+export function readTargetsFromPackageJson(packageJson: PackageJson) {
+  const { scripts, nx } = packageJson;
   const res: Record<string, TargetConfiguration> = {};
   Object.keys(scripts || {}).forEach((script) => {
     if (!nx?.includedScripts || nx?.includedScripts.includes(script)) {
       res[script] = buildTargetFromScript(script, nx);
     }
   });
 
-  // Add implicit nx-release-publish target for all package.json files to allow for lightweight configuration for package based repos
-  if (!res['nx-release-publish']) {
+  /**
+   * Add implicit nx-release-publish target for all package.json files that are
+   * not marked as `"private": true` to allow for lightweight configuration for
+   * package based repos.
+   */
+  if (!packageJson.private && !res['nx-release-publish']) {
```

Comment:
Do we have to consider `publishConfig` with `access: public`?

I don't think so right? `private` specifies if something can be published and access is only to choose between `restricted` and `public` right? But either way, `publishConfig` cannot disable publishing.

Reply:
Yes that’s right